web/src/routes/SurveyPages/SurveyFormPage.tsx

```diff
@@ -1,5 +1,400 @@
+    if (newOptions) {
+      newOptions[optionId] = { ...newOptions[optionId], [name]: value };
+    }
+    const newQuestionList = [...questionList];
+    newQuestionList[questionId] = { ...newQuestionList[questionId], questionOptions: newOptions };
+    setQuestionList(newQuestionList);
+  };
+
+  const handleClickButton = (event: React.MouseEvent<HTMLButtonElement>, questionId: number, optionId?: number) => {
+    const { name } = event.target as HTMLInputElement;
+    if (name === 'addQuestion') addQuestionUnderId(questionId);
+    else if (name === 'deleteQuestion') deleteQuestionAtId(questionId);
+    else if (name === 'addOption') addOptionAtBottom(questionId);
+    else if (typeof optionId !== 'undefined') deleteOptionAtId(questionId, optionId);
+  };
+
+  const makeSubjectiveAnswerForm = (questionId: number, selected: number) => {
+    return (
+      <QuestionContainer
+        ref={(element) => {
+          questionRefs.current[questionId] = element as HTMLDivElement;
+        }}
+        theme={theme}
+        key={questionId}
+      >
+        {SubjectiveAnswerForm({
+          selected,
+          questionId,
+          handleChangeQuestion,
+          handleChangeQuestionType,
+          questionList,
+          handleClickButton,
+          theme,
+        })}
+      </QuestionContainer>
+    );
+  };
+
+  const makeoptionsForm = (questionId: number) => {
+    const tmpOptions = questionList[questionId].questionOptions;
+    if (typeof tmpOptions !== 'undefined') {
+      return NumberUtils.range(0, tmpOptions.length).map((index: number) => (
+        <OptionContainer theme={theme} key={index}>
+          <OptionInput
+            theme={theme}
+            onChange={(event) => handleChangeOption(event, questionId, index)}
+            name="option"
+            value={tmpOptions[index].option || ''}
+          />
+          <DeleteOptionButton
+            theme={theme}
+            name="deleteOption"
+            onClick={(event) => handleClickButton(event, questionId, index)}
+          >
+            X
+          </DeleteOptionButton>
+        </OptionContainer>
+      ));
+    }
+    return <AnswerLable theme={theme}>옵션을 추가해 주세요</AnswerLable>;
+  };
+
+  const makeChoiceForm = (questionId: number, selected: number) => {
+    return (
+      <QuestionContainer
+        ref={(element) => {
+          questionRefs.current[questionId] = element as HTMLDivElement;
+        }}
+        theme={theme}
+        key={questionId}
+      >
+        {ChoiceAnswerForm({
+          selected,
+          questionId,
+          handleChangeQuestion,
+          handleChangeQuestionType,
+          questionList,
+          handleClickButton,
+          makeoptionsForm,
+          theme,
+        })}
+      </QuestionContainer>
+    );
+  };
+
+  const showQuestionForm = (questionType: number, questionId: number) => {
+    switch (questionType) {
+      case QuestionType.LONG_ANSWER || QuestionType.SHORT_ANSWER:
```

Comment:
TypeScript enum types are actually '`number`'s. So `QuestionType.LONG_ANSWER` doesn't check anything but it's bare number. And `||` is TypeScript boolean operator. We've handled on our previous Weekly Meeting. 

Please think again what would be returned as a result of `QuestionType.LONG_ANSWER || QuestionType.SHORT_ANSWER` operation. 

This is the reason why I've clicked on `SHORT_ANSWER` but still sees buttons to add options.